scripts/science_archive/ingest_sample_data.py

```diff
@@ -0,0 +1,217 @@
+obs_portal_ready = False
+while not obs_portal_ready:
+    try:
+        requests.get('http://observation-portal:8000/api/').text  # try connecting
+        obs_portal_ready = True
+    except Exception as e:
+        print("...waiting for the observation portal to come online...")
+        time.sleep(1)
+    
+# These credentials are defined in the observation-portal startup command in the 
+# docker compose file, during user creation.
+obs_portal_username = 'test_user'
+obs_portal_password = 'test_pass'
+
+# Trade the username/password for a token granting access to completed observations
+observations_token_endpoint = os.environ['OBSERVATION_PORTAL_BASE_URL'] + '/api-token-auth/'
+observations_token = requests.post(
+    observations_token_endpoint,
+    data = {
+        'username': obs_portal_username,
+        'password': obs_portal_password
+    }
+).json().get('token')
+
+
+################################################################################
+# Import the ingester only after the needed environment variables have been set.
+################################################################################
+from ocs_ingester import ingester
+
+
+################################################################################
+# Create a temporary directory for working with our fits file.
+# The goal is to run a loop that modifies our sample data's metadata to match
+# a completed observation, and then upload that file. 
+################################################################################
+
+# Make a temporary directory
+tmp_fits_dir =  f'/example_data/tmp_fits'
+try:
+    os.makedirs(tmp_fits_dir)
+except OSError:
+    print('Could not create the directory ', tmp_fits_dir)
+    pass
+
+# copy the fits file so that we can modify it without changing the original
+file = 'ogg0m406-kb27-20210720-0305-s91.fits.fz'
+original_fits_path = f'/example_data/{file}'
+copied_fits_path = f'/example_data/tmp_fits/{file}'
+shutil.copyfile(original_fits_path, copied_fits_path)
+filename_counter = -1
+
+################################################################################
+# Helpers to get completed observations so we can make matching data files
+################################################################################
+def get_completed_observations():
+    observations_endpoint = f"{os.getenv('OBSERVATION_PORTAL_BASE_URL')}/observations"
+    query_params = {
+        "state": "COMPLETED",
+        "user": "test_user",
+    }
+    observations_endpoint += "?" + urllib.parse.urlencode(query_params)
+    obs_auth_header = { "Authorization": f"Token {observations_token}" }
+
+    observations = requests.get(observations_endpoint, headers=obs_auth_header).json()
+    pprint(len(observations["results"]))
+    return observations["results"]
+
+def header_vals_from_observation(obs):
+    return {
+        #"RLEVEL": "",
+        "DAY_OBS": obs['end'].split('T')[0],
```

Comment:
This comment is for both this line and the line below. I don't think the ingester is generating the correct path in the minio bucket - it figures out the dayobs [like this](https://github.com/observatorycontrolsystem/ingester/blob/534c4b97f735ac08833a508879ba98a951643f07/ocs_ingester/utils/fits.py#L252) but these header keys are using underscores instead of the expected dashes. I believe this is the reason I am unable to download a file from the bucket - with the stack up, go to http://localhost:9500/frames/5/, and then click on the url field value. I get this:

```
<Error>
<Code>NoSuchVersion</Code>
<Message>The specified version does not exist.</Message>
<Key>ogg/xx04/20210727/processed/ogg1m0a-xx04-20210727-0004-e91.fits.fz</Key>
<BucketName>ocs-example-bucket</BucketName>
<Resource>/ocs-example-bucket/ogg/xx04/20210727/processed/ogg1m0a-xx04-20210727-0004-e91.fits.fz</Resource>
<Region>minio-region</Region>
<RequestId>1699FEC2959F8538</RequestId>
<HostId>777bed38-739e-4053-af5b-0bb26290ae1d</HostId>
</Error>
```
And it looks like in the bucket itself (when browsing through the interface at port 9000), the dayobs was set to 20210720, and the file is actually located in http://localhost:9000/minio/ocs-example-bucket/ogg/xx04/20210720/processed/ but I would expect it to be in dayobs 20210727.